inbm/dispatcher-agent/dispatcher/sota/sota.py

```diff
@@ -125,7 +125,7 @@ def __init__(self,
             # If an exception occurs during string conversion, raise that exception
             raise SotaError('package_list is not a string in manifest') from e
 
-        self._device_reboot = parsed_manifest['deviceReboot']
+        self._device_reboot = str(parsed_manifest['deviceReboot'])
```

Comment:
**Suggestion:** Ensure that the deviceReboot key exists in parsed_manifest before attempting to access it to avoid a KeyError. [bug]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        self._device_reboot = str(parsed_manifest.get('deviceReboot', 'Yes'))
```